kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeResultGenerator.kt

```diff
@@ -254,17 +280,26 @@ data class TypeResultGenerator(val options: Options, val implicitAccessor: Acces
             if (args.isNotEmpty()) {
                 add(" ->")
             }
+            add("\n⇥")
 
-            add("%M(", name)
-            parameters.forEachIndexed { i, param ->
+            add("%M", name)
+            val firstParam = parameters.firstOrNull()?.result
+            if (firstParam is TypeResult.Function || firstParam is TypeResult.NamedFunction) {
```

Comment:
ditto, I wonder if there's a way to 'push' this logic into generating code for `TypeResult.Function`/`TypeResult.NamedFunction`.